### ISSUE TITLE
[TF 2.2] keras fit class weights are being cast to int

### ISSUE THREAD

**Issue by mgmverburg** [tensorflow/tensorflow#37747]
**System information** 
- Have I written custom code (as opposed to using a stock
example script provided in TensorFlow): No
- TensorFlow version : introduced since 2.2.0-rc0
- Python version: - Bazel
version (if compiling from source): Python 3 (Colab)

Google Colaboratory
Tensorflow 2.2.0-rc0 with Python 3

**Describe the current behavior**
The class weights passed to keras model.fit are converted to int64, which leads to all class weights that are below 1 being set to 0.

**Describe the expected behavior**
That the class weights are kept as float values, so that results can be more precise based on the class weights

**Other info / logs**
I tracked the piece of code that is responsible for this behavior: 
[https://github.com/tensorflow/tensorflow/blob/c0306ef626b02ab5ab10aac2cec6d08f56136a5c/tensorflow/python/keras/engine/data_adapter.py#L1258](url)
`class_weight_tensor = ops.convert_to_tensor_v2(
      [class_weight[c] for c in class_ids])	      [int(class_weight[c]) for c in class_ids], dtype="int64")`

and it seems that this commit for TF 2.2 introduced it, but I cannot figure out why it was changed like this:
[https://github.com/tensorflow/tensorflow/commit/10666c59dd4858645d1b03ce01f4450da80710ec#diff-f8dd40712ac721c1b363e1a1ec44c1a3](url)

As a temporary workaround, I tried to make my lowest weight normalized to 1, but then I still get a floating poant value for the other weight, so even if I round it, it will still not be as precise as I need it.
Since I use tf.dataset loaded from GCS, I cannot simply pass the sample_weights directly myself either as a workaround.

So I am not sure whether is was a bug, or an intended change. In case it was an intended change, I would like to know what the alternative would be.

**Comment by gadagashwini** [comment]
@mgmverburg, Could you provide the simple standalone code to replicate the reported issue. Thanks!

**Comment by mgmverburg** [comment]
@gadagashwini I can try and make a snippet with a simple model etc. but I don't think it will show much, because it isn't an error that is thrown, just incorrect behavior which can be directly explained by the casting to int. So somehow we will need to find out what the rationale was behind that change, because quite likely there must have been a motivation for that change, and changing it back might hence have negative side-effects.

But if you think it will help, I will try and make a standalone snippet later today.

**Comment by goldiegadde** [comment]
@mgmverburg this [commit](https://github.com/tensorflow/tensorflow/commit/bfc51b2d2384a6945ce7023aa08f6c629766ebdf) should fix this issue. 
the fix has also been cherrypicked into r2.2 branch.

**Comment by mgmverburg** [comment]
> @mgmverburg this [commit](https://github.com/tensorflow/tensorflow/commit/bfc51b2d2384a6945ce7023aa08f6c629766ebdf) should fix this issue.
> the fix has also been cherrypicked into r2.2 branch.

As far as I can tell, it is fixed! Thank you!

**Comment by jvishnuvardhan** [comment]
Closing this as this was resolved. Thanks!
